Functions/xASL_test_Flavors.m

```diff
@@ -1,26 +1,27 @@
-function xASL_test_Flavors(pathExploreASL, pathTest, bTest, x)
+function flavors = xASL_test_Flavors(testConfig, bTest, x, flavors)
 %xASL_test_Flavors Runs the complete testing of Flavors including import from DICOM to BIDS, processing and comparison
 %
-% FORMAT: xASL_test_Flavors(pathExploreASL, pathTest[, bTest, x])
+% FORMAT: xASL_test_Flavors(pathExploreASL, pathFlavorDatabase[, bTest, x])
 %
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
 % 
 % INPUT:
-%   pathExploreASL - full path to the ExploreASL code (REQUIRED)
-%   pathTest       - full path to the testing directory containing the FlavorsDatabase with all the data (REQUIRED)
-%   bTest          - an array of booleans specifying which subparts of the test are supposed to be run
-%                    1. Make a copy of the flavors data
-%                    2. Run the DCM->BIDS import
-%                    3. Check the DCM->BIDS import results
-%                    4. Run BIDS->Legacy import
-%                    5. Check the the BIDS->Legacy import results
-%                    6. Run the ExploreASL on all datasets
-%                    7. Checks the ExploreASL processing results
-%                    (OPTIONAL, DEFAULT = [1 1 1 1 1 1 1])
-%   x              - x structure (OPTIONAL, DEFAULT = run Initialization)
+%   testConfig         - struct which contains the paths to the ExploreASL
+%                        installation and the testing/flavor repository (REQUIRED) 
+%   bTest              - an array of booleans specifying which subparts of the test are supposed to be run
+%                        1. Make a copy of the flavors data
+%                        2. Run the DCM->BIDS import
+%                        3. Check the DCM->BIDS import results
+%                        4. Run BIDS->Legacy import
+%                        5. Check the the BIDS->Legacy import results
+%                        6. Run the ExploreASL on all datasets
+%                        7. Checks the ExploreASL processing results
+%                        (OPTIONAL, DEFAULT = [1 1 1 1 1 1 1])
```

Comment:
The last 3 inputs are obviously not optional...
It doesn't work to run
xASL_test_Flavors(spm_jsonread('/home/janpetr/ExploreASL/ExploreASL/Testing/testConfig.json'),[],[],[])

We need to correctly check and initialized the defaults.

Reply:
Yep, so like above, I never had these problems since I mainly tested by running `xASL_test_FullPipelineTest`. I understand that this is not 100% correct though.
Does this 04b6f5a fix it already? I describe the last input as recommended and there is a fallback with a warning.

Reply:
x should be optional, there is a check for that.

Reply:
Same goes for bTest, I think you implemented that.